cmd/appliance/switch_partition.go

```diff
@@ -0,0 +1,225 @@
+			if err != nil {
+				return err
+			}
+			opts.api = api
+
+			ctx := context.Background()
+			if opts.id == nil {
+				id, err := appliancepkg.PromptSelect(ctx, api, nil, nil, false)
+				if err != nil {
+					return err
+				}
+				uid, err := uuid.Parse(id)
+				if err != nil {
+					return err
+				}
+				opts.id = &uid
+			}
+
+			if opts.id == nil {
+				return fmt.Errorf("failed to switch partition: no appliance identifier provided")
+			}
+
+			stats, _, err := api.Stats(ctx, nil, nil, false)
+			if err != nil {
+				return err
+			}
+			for _, s := range stats.GetData() {
+				if opts.id.String() != s.GetId() {
+					continue
+				}
+				opts.applianceStats = &s
+			}
+
+			minVersion, _ := version.NewVersion("6.2.6-0")
+			currentVersion, err := version.NewVersion(opts.applianceStats.GetVersion())
+			if err != nil {
+				return err
+			}
+			i, err := appliancepkg.CompareVersionsAndBuildNumber(minVersion, currentVersion)
+			if err != nil {
+				return err
+			}
+			if i < 0 {
+				return fmt.Errorf("minimum supported version for the 'switch-partition' command is 6.2.6. current version is %s", currentVersion.String())
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return switchPartitionRunE(opts)
+		},
+	}
+
+	return cmd
+}
+
+func switchPartitionRunE(opts *options) error {
+	ctx := context.Background()
+	api := opts.api
+	if api.ApplianceStats == nil {
+		api.ApplianceStats = &appliancepkg.ApplianceStatus{
+			Appliance: api,
+		}
+	}
+
+	appliance, err := api.Get(ctx, opts.id.String())
+	if err != nil {
+		return fmt.Errorf("failed to get appliance: %w", err)
+	}
+
+	volume := opts.applianceStats.GetVolumeNumber()
+	log := logrus.WithFields(logrus.Fields{
+		"id":     opts.id.String(),
+		"volume": volume,
+	})
+	log.Info("initial appliance stats")
+
+	summary, err := switchPartitionSummary(appliance.GetName())
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(opts.out, summary)
+	if opts.canPrompt {
+		if err := prompt.AskConfirmation(); err != nil {
+			return err
+		}
+	}
+
+	var p *tui.Progress
+	var t *tui.Tracker
+	if !opts.ciMode {
+		p = tui.New(ctx, opts.spinnerOut())
+		t = p.AddTracker(appliance.GetName(), opts.applianceStats.GetState(), "finished")
+		go t.Watch(appliancepkg.StatusNotBusy, []string{"error"})
+	}
+
+	err = backoff.Retry(func() error {
+		return api.ApplianceSwitchPartition(ctx, opts.id.String())
+	}, backoff.NewExponentialBackOff())
+	if err != nil {
+		return fmt.Errorf("partition switch failed on appliance %s: %v", opts.id, err)
+	}
+
+	if err := api.ApplianceStats.WaitForApplianceState(ctx, *appliance, appliancepkg.StatReady, t); err != nil {
+		if t != nil {
+			t.Fail(err.Error())
+		}
+		return fmt.Errorf("partition switch failed: %w", err)
+	}
+
+	if p != nil {
+		p.Wait()
+	}
+
+	// verify partition switch
+	stats, _, err := api.Stats(ctx, nil, nil, false)
+	if err != nil {
+		return fmt.Errorf("partition switch failed: %w", err)
+	}
```

Comment:
This error message might be a little misleading. It's not the switch-partition failing but the status call failing